src/include/planner/plannodes/create_table_plan_node.h

```diff
@@ -607,8 +607,8 @@ class CreateTablePlanNode : public AbstractPlanNode {
       if (col->GetCheckExpression()->GetReturnValueType() == type::TypeId::BOOLEAN) {
         check_cols.push_back(col->GetColumnName());
 
-        parser::ConstantValueExpression *const_expr_elem =
-            dynamic_cast<parser::ConstantValueExpression *>(col->GetCheckExpression()->GetChild(1).get());
+        auto const_expr_elem =
+            col->GetCheckExpression()->GetChild(1).CastManagedPointerTo<const parser::ConstantValueExpression>();
```

Comment:
I'm pretty sure this incorporates two undocumented and potentially invalid assumptions:
1. This is assumes the check expression has at least two children (not enforced anywhere).  My guess is that someone is assuming that it will always be an operator.  However, what if it is a UDF?
2. This assumes that the second argument must be constant.  I'm pretty sure column constraints can be arbitrary boolean expressions.

At a minimum the assumptions and their justifications need to be documented, but I think the real solution is probably storing the expression in `check_info` and not transient values.  I've been working on the assumption that constraints would be described by expressions as I write the catalog for the same reason that default values and index key descriptions are expressions and not transient values.